code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm

```diff
@@ -55,10 +55,14 @@
 		piping_layer = board.pipe_layer
 		set_layer = piping_layer
 
+	..() //Skipping the rest of on_construction() would be a bad idea so we clean up after it instead.
+
 	if(check_pipe_on_turf())
-		deconstruct(TRUE)
-		return
-	return..()
+		set_anchored(FALSE)
+		set_panel_open(TRUE)
+		change_pipe_connection(TRUE)
+		icon_state = "thermo-open"
+		visible_message(span_warning("You fail to anchor the thermomachine as it's port is being hogged up by something else."))
```

Comment:
this should be a balloon alert, right now not only is it a pointless chat, but it's also being sent to everyone as "you"

Reply:
Should I change the other visible message as well?

Reply:
I've made it no longer address people as "you", but I'm also not sure how I'm meant to turn it into a balloon alert without a mob to send it to. It's also a pretty long message and I have no idea on how to convey it other than just "Anchor failed" which sounds like shit.